core/client/src/main/java/alluxio/client/UnderFileSystemFileWriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.util.CommonUtils;
+
+import com.google.common.base.Throwables;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * The interface to write an under file system file through a worker's data server.
+ */
+public interface UnderFileSystemFileWriter extends Closeable {
+
+  /**
+   * The factory for the {@link UnderFileSystemFileWriter}.
+   */
+  class Factory {
+
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory for {@link UnderFileSystemFileWriter}.
+     *
+     * @return a new instance of {@link UnderFileSystemFileWriter}
+     */
+    public static UnderFileSystemFileWriter create() {
+      try {
+        return CommonUtils.createNewClassInstance(
+            Configuration.<UnderFileSystemFileWriter>getClass(
+                PropertyKey.USER_UFS_FILE_WRITER_CLASS), null, null);
+      } catch (Exception e) {
+        throw Throwables.propagate(e);
+      }
+    }
+  }
+
+  /**
+   * Writes data to the file in the under file system.
+   *
+   * @param address worker address to write the data to
+   * @param ufsFileId worker file id referencing the file
+   * @param fileOffset the offset in bytes where in the file to start writing, only sequential
+   *                   writes are supported
+   * @param bytes data to write
+   * @param offset start offset in bytes of the data to write
+   * @param length length to write in bytes
+   * @throws IOException if an error occurs during the write
+   */
+  void write(InetSocketAddress address, long ufsFileId, long fileOffset, byte[] bytes,
```

Comment:
how about changing `byte[] bytes` to `byte[] buffer` or `byte[] source`? otherwise, the text "in bytes" in above javadoc is confusing as it can be interpreted in different ways.